app/ui/src/app/support/support.component.ts

```diff
@@ -0,0 +1,158 @@
+  styleUrls: ['./support.component.scss']
+})
+export class SupportComponent implements OnInit {
+
+  // Used to bind patternfly behavior with foreign form elements
+  @ViewChild('alllogs') allogs: ElementRef;
+  @ViewChild('specificlogs') specificlogs: ElementRef;
+
+  allLogsSelected : boolean = true;
+  loading : boolean= true;
+
+  private restangularService: Restangular;
+
+  filter: ObjectPropertyFilterConfig = {
+    filter: '',
+    propertyName: 'name'
+  };
+
+  // List configuration
+  listConfig = {
+    multiSelect: true,
+    selectItems: false,
+    showCheckbox: true
+  };
+
+  // Toolbar configuration
+  toolbarConfig = {
+    filterConfig: {
+      fields: [
+        {
+          id: 'name',
+          title: 'Name',
+          placeholder: 'Filter by Name...',
+          type: 'text'
+        },
+        {
+          id: 'description',
+          title: 'Description',
+          placeholder: 'Filter by Description...',
+          type: 'text'
+        }
+      ]
+    }
+  };
+
+  items: Array<any> = [
+    {
+        name: "Integration Name 1",
+        description: "Description about the integration goes here"
+    },
+    {
+        name: "Integration Name 2",
+        description: "Description about the integration goes here"
+    }
+  ];
+
+  constructor(
+    public store: IntegrationStore,
+    public integrationSupportService: IntegrationSupportService,
+  ) {
+  }
+  
+  buildData(data: any = {}) {
+    console.log(data);
+    this.integrationSupportService
+      .downloadSupportData(data)
+      .subscribe(response => {
+        fileSaver.saveAs(response, "syndesis.zip");
+      },
+      error => {
+        console.log("Error downloading the file.");
+        console.log(error);
+      }
+    );
+    return {  };
+  }
+
+  // Handles events when the user interacts with the toolbar filter
+  filterChanged($event) {
+    // TODO update our pipe to handle multiple filters
+    if ($event.appliedFilters.length === 0) {
+      this.filter.filter = '';
+    }
+    $event.appliedFilters.forEach(filter => {
+      this.filter.propertyName = filter.field.id;
+      this.filter.filter = filter.value;
+    });
+  }
+
+  onSubmit() {
+    let chosen = [];
+    if(this.allLogsSelected) {
+      chosen = this.items;
+    } else {
+      chosen = this.items.filter(x => x.selected === true);
+    }
+    let input = {};
+    chosen.forEach(el => input[el.name] = true);
+    console.log("integrations: " + JSON.stringify(chosen));
+    this.buildData(input);
+  }
+
+  deselectAll() {
+    this.items.forEach(item => item.selected = false);
+  }
+
+  handleSelectionChange(event) {
+    this.allLogsSelected = false;
+  }
+
+  totalItems() : number {
+    return this.items.length;
+  }
+
+  selectedItems() : number {
+    return this.items.filter(x => x.selected === true).length;
+  }
+
+  public ngOnInit() {
+    this.store.list.subscribe(integration  => {
+      this.items = integration as Integration[];
+    });
```

Comment:
A few remarks here:

- All class members in TypeScript are `public` by default, so we never annotate that accessor. The `public` keyword is used in constructor parameters only, to inform the compiler that such parameter also must become a class member.
- You should not typecast the value in the assignation expression, but on the lambda declaration instead,
- Code should be self-explanatory, hence lambda names should be too. If `integration` is meant to be an array of `Integration` objects (therefore a collection) then name it accordingly as `integrations`. There's actually a shortcut type name `Integrations` defined at a model level for this, so you can use it instead.
- This subscribe statement will keep emitting data long after this component is destroyed, because that is teh nature of the `Observer` pattern. Therefore, the `Subscription` should be disposed upon tearing down the component. Please implement the `OnDestroy` interface and add an `unsubscribe` line to it.